components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java

```diff
@@ -0,0 +1,332 @@
+        AzureADRequestInterceptor appInterceptor = new AzureADRequestInterceptor(accessTokenGenerator);
+
+        appClient = this.buildFeignClient(new OkHttpClient(), appInterceptor)
+                .target(ApplicationClient.class, graphApiEndpoint);
+    }
+
+    private Builder buildFeignClient(
+            OkHttpClient client, AzureADRequestInterceptor interceptor) {
+        return Feign.builder()
+                .client(client)
+                .requestInterceptor(interceptor)
+                .encoder(new GsonEncoder())
+                .decoder(new GsonDecoder())
+                .errorDecoder(new KMClientErrorDecoder())
+                .logger(new Slf4jLogger());
+    }
+
+    @Override
+    public OAuthApplicationInfo createApplication(OAuthAppRequest oauthAppRequest) throws APIManagementException {
+        OAuthApplicationInfo oauthAppInfo = oauthAppRequest.getOAuthApplicationInfo();
+
+        if (oauthAppInfo != null) {
+            ClientInformation appInfo = this.getClientInformation(oauthAppInfo);
+            ClientInformation app;
+
+            try {
+                app = appClient.createApplication(appInfo);
+                if (app != null) {
+                    PasswordInfo pInfo = this.setPassword(app.getId());
+                    app.setClientSecret(pInfo.getSecret());
+                    return getOAuthApplicationInfo(app);
+                }
+            } catch (KeyManagerClientException e) {
+                handleException("Error occurred while creating Azure AD Application", e);
+            }
+            return oauthAppInfo;
+        }
+
+        return null;
+    }
+
+    private PasswordInfo setPassword(String id) throws KeyManagerClientException {
+        PasswordCredential passwordCredential = new PasswordCredential();
+        passwordCredential.setDisplayName("app_secret");
+
+        PasswordInfo passwordInfo = new PasswordInfo();
+        passwordInfo.setPasswordCredential(passwordCredential);
+
+        return appClient.addPassword(id, passwordInfo);
+    }
+
+    private OAuthApplicationInfo getOAuthApplicationInfo(ClientInformation appInfo) {
+        OAuthApplicationInfo oauthAppInfo = new OAuthApplicationInfo();
+        oauthAppInfo.setClientName(appInfo.getAppName());
+        oauthAppInfo.setClientId(appInfo.getClientId());
+        oauthAppInfo.setClientSecret(appInfo.getClientSecret());
+
+        oauthAppInfo.addParameter(ApplicationConstants.OAUTH_CLIENT_GRANT,
+                AzureADConstants.CLIENT_CREDENTIALS_GRANT_TYPE);
+
+        // Add object UUID
+        oauthAppInfo.addParameter(AzureADConstants.OBJECT_ID, appInfo.getId());
+
+        return oauthAppInfo;
+    }
+
+    private ClientInformation getClientInformation(OAuthApplicationInfo oauthAppInfo) {
+        ClientInformation clientInformation = new ClientInformation();
+        clientInformation.setAppName(oauthAppInfo.getClientName());
+        String id = (String) oauthAppInfo.getParameter(AzureADConstants.OBJECT_ID);
```

Comment:
You have done a null check for `id` in the next line. One possible scenario for the id to become null is `oauthAppInfo.getParameter(AzureADConstants.OBJECT_ID)` getting null. If that happens there will be an exception when casting to String in this line. You have to check whether that property exists before casting to string